bridge/src/bridge/withdraw_relay.rs

```diff
@@ -113,6 +113,7 @@ impl<T: Transport> Stream for WithdrawRelay<T> {
 	fn poll(&mut self) -> Poll<Option<Self::Item>, Self::Error> {
 		let app = &self.app;
 		let gas = self.app.config.txs.withdraw_relay.gas.into();
+		let gas_price = self.app.gas_prices.lock().unwrap().get_home_price();
```

Comment:
The biggest problem I see with the approach taken in `GasPriceClient` is that every timeout period (which is set to 10 seconds currently) this will lock up the event loop for the duration of the request (which is not zero). Blending this functionality in properly would rather require having a Stream underneath so that it's polled until it produces a new value (which it does every X seconds -- this can be coupled with `tokio-timer` to produce the intended result)

Reqwest's async functionality is unstable and is under feature flag: https://docs.rs/reqwest/0.8.5/reqwest/unstable/async/index.html

I think there are other crates that provide futures-based HTTP clients, as well.